core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -692,7 +693,7 @@ private void maybeRefreshEngine() {
                             if (shard.isRefreshNeeded()) {
                                 shard.refresh("schedule");
                             }
-                        } catch (EngineClosedException | AlreadyClosedException ex) {
+                        } catch (IndexShardClosedException | AlreadyClosedException ex) {
```

Comment:
`EngineClosedException` inherits from `IndexShardClosedException` (and yes it can be thrown)